User can register

swardbladpaulo opened the pull request:
[API provides functionality to create user account ](https://www.pivotaltracker.com/n/projects/2478349)
### User Story
```
As an API
In order for users to search through quotes
We would like to authenticate the visitor
```
## Changes proposed in this pull request:
* add new specfile for login in, change name on spec for registration, added settings in the devise token auth initializer
* Creating api folder in spec/requests folder,
* Moving request spec files to spec/requests/api folder, Creating a test for user registration
* Renaming the spec file from get_five_random_trump_quotes to get_a_random_trump_quote.
* Modifying Devise route and making Devise to skip omniauth_callbacks.
* Adding devise_token_auth
* Creating a User model
* Creating a factory bot for users
* Creating a model spec for user model
* Adding and configuring rack-cors.

## Packages/Gems added
* rack-cors
* devise token_auth

KyunginNa committed:
Adding webmock and Creating a fixture file for random quote.

KyunginNa committed:
Resolving merge conflict

KyunginNa committed:
Resolving conflict

KyunginNa committed:
Resolving conflicts

swardbladpaulo committed:
troubleshoot deploy from backend

swardbladpaulo committed:
testing

KyunginNa committed:
Adding rack-cors in Gemfile and bundle install.

swardbladpaulo committed:
trobleshooting

swardbladpaulo committed:
accept incoming changes

swardbladpaulo committed:
Merge branch 'show_a_random_quote' of https://github.com/swardbladpaulo/trumpster_api into development

KyunginNa committed:
Merge branch 'development' of https://github.com/CraftAcademy/trumpster_api into development

swardbladpaulo committed:
Merge branch 'development' of https://github.com/CraftAcademy/trumpster_api into development

KyunginNa committed:
Adding and configuring rack-cors.

KyunginNa committed:
Modifying rack-cors config to use devise-token-auth.

KyunginNa committed:
Adding devise_token_auth and run bundle install
Creating a User model
Creating a factory bot for users
Creating a model spec for user model

KyunginNa committed:
Modifying Devise route and making Devise to skip omniauth_callbacks.

KyunginNa committed:
Renaming the spec file from get_five_random_trump_quotes to get_a_random_trump_quote.

KyunginNa committed:
Creating api folder in spec/requests folder,
Moving request spec files to spec/requests/api folder,
Creating a test for user registration.

swardbladpaulo committed:
Merge branch 'userCanRegister' of https://github.com/KyunginNa/trumpster_api into userCanRegister

swardbladpaulo committed:
add new specfile for login in, change name on spec for registration, added settins in the devise token auth initializer

tochman commented on app/controllers/api/quotes_controller.rb:
```diff
@@ -1,15 +1,14 @@
 class Api::QuotesController < ApplicationController
-  API_URL = 'https://api.tronalddump.io'.freeze
+  API_URL = 'https://api.tronalddump.io'
```
Don’t remove the .freeze here.

KyunginNa replied:
Yes, we put it back

tochman commented on app/controllers/api/quotes_controller.rb:
```diff
@@ -1,15 +1,14 @@
 
   def show
-    if params[:id] == 'random'
+    if params[:id] == "random"
```
In ruby, use single quotes unless you string interpolate

KyunginNa replied:
We changed them to single quotes.

tochman commented on app/controllers/application_controller.rb:
```diff
@@ -1,2 +1,3 @@
 class ApplicationController < ActionController::API
+        include DeviseTokenAuth::Concerns::SetUserByToken
```
Please format your code

KyunginNa replied:
Yes, this is handled now.

tochman commented on app/models/user.rb:
```diff
@@ -0,0 +1,10 @@
+class User < ActiveRecord::Base
+  extend Devise::Models
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable and :omniauthable
```
Please remove comments

KyunginNa replied:
They are deleted now.

tochman commented on spec/requests/api/registrations_spec.rb:
```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+RSpec.describe "GET /api/auth", type: :request do
```
You are testing a POST request

KyunginNa committed:
Added freeze
Changed double quotes to single ones in quote_controller,
Formatted code
Removed unnecessary comments
